app/test/prepare_proposal_context_test.go

```diff
@@ -0,0 +1,102 @@
+package app_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+// TestTimeInPrepareProposalContext checks for an edge case where the block time
+// needs to be included in the sdk.Context that is being used in the
+// antehandlers. If a time is not included in the context, then the second
+// transaction in this test will always be filtered out, result in vesting
+// accounts never being able to spend funds.
+func TestTimeInPrepareProposalContext(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping DebugVestingAccountsTime test in short mode.")
+	}
+	accounts := make([]string, 35)
+	for i := 0; i < len(accounts); i++ {
+		accounts[i] = tmrand.Str(9)
+	}
+	cfg := testnode.DefaultConfig().WithAccounts(accounts)
+	cctx, _, _ := testnode.NewNetwork(t, cfg)
+	ecfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
+	type test struct {
+		name         string
+		msgFunc      func() (msgs []sdk.Msg, signer string)
+		expectedCode uint32
+	}
+	tests := []test{
+		{
+			name: "create continuous vesting account with a start time in the future",
+			msgFunc: func() (msgs []sdk.Msg, signer string) {
+				vestAccName := "vesting"
```

Comment:
[optional] I think this could be hoisted up to line ~38 and removed from this line and the second test case b/c IIUC the `vestAccName` should be identical for both test cases.

Reply:
https://github.com/celestiaorg/celestia-app/pull/2515/commits/5ad837147e243a008735eb1faaa5679240a7b2c4